projects/pt1/python/torch_mlir_e2e_test/registry.py

```diff
@@ -38,6 +40,7 @@ def decorator(f):
                 unique_name=f.__name__,
                 program_factory=module_factory,
                 program_invoker=f,
+                timeout=timeout,
```

Comment:
This is error prone without units. Minimally, include units, so `timeout_s` or `timeout_seconds`.

Alternately, use a unitless object that represents a duration like `timedelta`: https://docs.python.org/3/library/datetime.html#timedelta-objects. That can be converted to any units that you want at the location that units are needed. If you need to convert to different units or perform math with the duration, the object can handle that for you with no risk of using the wrong number of decimal places.

See also https://testing.googleblog.com/2017/11/

Reply:
great point. using timeout_seconds.

I personally love the timedelta idea but also i think in this case simplicity trumps elegance / adaptability